### PR TITLE
User friends

### PR DESCRIPTION
**Pull request by tanishk15sharma**
- [x] Style : friends list section
- [x] get followers and following
- [x] `useState` to display followers and following section
- [x] follow/unfollow Rest API call

### CODE REVIEW

File: src/components/UserFriends.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        {showFriends
+          ? myFollowersList?.map((friend) => (
+              <div
+                className="flex flex-col flex-wrap justify-center"
+                key={friend._id}
+              >
+                <Link to={`/profile/${friend._id}`}>
+                  <div className="w-24 h-24 bg-primary-200 rounded-full flex  justify-center items-center font-bold text-primary-900 text-4xl">
+                    {friend.name && friend.name[0].toUpperCase()}
+                    {/* <img src={profileBg} alt="profile-bg" /> */}
```
Comment (znjs) [inline src/components/UserFriends.js:47]: can remove the commented code

File: src/features/authSlice.js [hunk 2]
```diff
@@ -39,34 +39,26 @@ const postSignupDetails = createAsyncThunk(
...
-  following: [],
-};
 const authSlice = createSlice({
   name: "authentication",
-  initialState,
+  initialState: {},
   reducers: {},
   extraReducers: {
     [postLoginDetails.pending]: () => {
       console.log("login pending");
```
Comment (znjs) [inline src/features/authSlice.js:48]: can remove the `console.log`

File: src/features/authSlice.js [hunk 3]
```diff
@@ -39,34 +39,26 @@ const postSignupDetails = createAsyncThunk(
...
+      console.log(payload.data);
+      state = payload.data.data.user;
+      // state.username = payload.data.user.username;
+      // state.name = payload.data.user.name;
+      // state.bio = payload.data.user.bio;
+      // state.profileImage = payload.data.user.profileImage;
+      // state.profileCover = payload.data.user.profileCover;
+      // // state.bookmark = payload.user.bookmarks;
+      // state.followers = payload.data.user.followers;
+      // state.following = payload.data.user.following;
```
Comment (znjs) [inline src/features/authSlice.js:60]: can remove the commented code

File: src/features/authSlice.js [hunk 4]
```diff
@@ -39,34 +39,26 @@ const postSignupDetails = createAsyncThunk(
...
-      state.username = payload.data.user.username;
-      state.name = payload.data.user.name;
-      state.bio = payload.data.user.bio;
-      state.profileImage = payload.data.user.profileImage;
-      state.profileCover = payload.data.user.profileCover;
-      // state.bookmark = payload.user.bookmarks;
-      state.followers = payload.data.user.followers;
-      state.following = payload.data.user.following;
-      localStorage?.setItem("userToken", payload.data.token);
+      console.log(payload.data);
```
Comment (znjs) [inline src/features/authSlice.js:51]: remove the console.log

File: src/features/authSlice.js [hunk 5]
```diff
@@ -76,15 +68,17 @@ const authSlice = createSlice({
...
+      state = payload.data;
+
+      // state.username = payload.user.username;
+      // state.name = payload.user.name;
+      // state.bio = payload.user.bio;
+      // state.profileImage = payload.user.profileImage;
+      // state.profileCover = payload.user.profileCover;
+      // // state.bookmark = payload.user.bookmarks;
+      // state.followers = payload.user.followers;
+      // state.following = payload.user.following;
```
Comment (znjs) [inline src/features/authSlice.js:80]: can remove the commented code

File: src/components/UserFriends.js [hunk 6]
```diff
@@ -54,7 +58,22 @@ const UserFriends = ({ user }) => {
...
-          : null}
+          : myFollowingList?.map((friend) => (
+              <div
+                className="flex flex-col flex-wrap justify-center"
+                key={friend._id}
+              >
+                <Link to={`/profile/${friend._id}`}>
+                  <div className="w-24 h-24 bg-primary-200 rounded-full flex  justify-center items-center font-bold text-primary-900 text-4xl">
+                    {friend.name && friend.name[0].toUpperCase()}
+                    {/* <img src={profileBg} alt="profile-bg" /> */}
```
Comment (sharathnair9999) [inline src/components/UserFriends.js:63]: remove if it is not needed

File: src/utils/user.js [hunk 7]
```diff
@@ -9,4 +10,61 @@ const getUser = async (id) => {
   }
 };
 
-export { getUser };
+const getUserFollowing = async (userId) => {
+  if (userId) {
+    try {
+      const { data, status } = await axios.get(`/users/myFollowing/${userId}`);
+      console.log(data);
```
Comment (znjs) [inline src/utils/user.js:17]: can remove the console

File: src/utils/user.js [hunk 8]
```diff
@@ -9,4 +10,61 @@ const getUser = async (id) => {
...
+      if (status === 200) {
+        return data.followingList;
+      }
+    } catch (err) {
+      console.log(err);
```
Comment (znjs) [inline src/utils/user.js:22]: can change the errors to `console.error`

**Review by sharathnair9999 — APPROVED** [review]
Looks good.

**Review by znjs — APPROVED** [review]
- Good PR
- Good naming conventions

File: src/components/UserFriends.js (hunk 6 above)
Comment (tanishk15sharma) [reply src/components/UserFriends.js:63]: yes , I need it later .

File: src/features/authSlice.js (hunk 3 above)
Comment (tanishk15sharma) [reply src/features/authSlice.js:60]: yes , thank you for suggesting  . I will do the changes

**Pull request merged**